app/src/main/java/com/android/unio/ui/components/NotificationSender.kt

```diff
@@ -0,0 +1,146 @@
+package com.android.unio.ui.components
+
+import android.widget.Toast
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.android.unio.R
+import com.android.unio.model.notification.NotificationType
+import com.android.unio.model.notification.broadcastMessage
+import com.android.unio.model.strings.test_tags.NotificationSenderTestTags
+import com.android.unio.model.utils.TextLength
+import com.android.unio.ui.theme.AppTypography
+
+/**
+ * A composable that shows a dialog for sending a notification to a topic.
+ *
+ * @param dialogTitle The title of the dialog.
+ * @param notificationType The target of the notification.
+ * @param topic The topic to send the notification to.
+ * @param notificationContent A function that returns the notification content given the written
+ *   message.
+ * @param showNotificationDialog Whether to show the dialog.
+ * @param onClose The function to call when the dialog is closed.
+ */
+@Composable
+fun NotificationSender(
```

Comment:
Can I please finish a review before you modify the code ? @oskar-codes

Reply:
So as i said, good job on fixing the : 
- message that did not clear if you exit the dialog by tapping out
- extracting the hardcoded the text constant
- adding some padding :+1: